### ISSUE TITLE
[BUG] Bento components on NEXTJS - ReferenceError: self is not defined

### ISSUE THREAD

**Issue by Beethoven** [ampproject/bentojs.dev#183]
Following guide at

https://bentojs.dev/documentation/guides/bento-next.js-guide/

and gets this error:
ReferenceError: self is not defined

**Comment by AnuragVasanwala** [comment]
Hi @Beethoven 👋🏻

Would you please try using dynamic `import` with `ssr` disabled instead of importing conventionally?

Syntax:
```js
import dynamic from "next/dynamic";

const <module_name> = dynamic(() =>
  import(<path>).then((mod) => mod.<component_to_use>),
  { ssr: false }
);
```

So, this import block:
```js
import {
  BentoAccordion,
  BentoAccordionSection,
  BentoAccordionHeader,
  BentoAccordionContent,
} from '@bentoproject/accordion/react';

import '@bentoproject/accordion/styles.css';
```
would be written as:
```js
import dynamic from "next/dynamic";

// Use dynamic import with SSR disabled for all imports except CSS
const BentoAccordion = dynamic(() =>
  import("@bentoproject/accordion/react").then((mod) => mod.BentoAccordion),
  { ssr: false }
);
const BentoAccordionSection = dynamic(() =>
  import("@bentoproject/accordion/react").then((mod) => mod.BentoAccordionSection),
  { ssr: false }
);
const BentoAccordionHeader = dynamic(() =>
  import("@bentoproject/accordion/react").then((mod) => mod.BentoAccordionHeader),
  { ssr: false }
);
const BentoAccordionContent = dynamic(() =>
  import("@bentoproject/accordion/react").then((mod) => mod.BentoAccordionContent),
  { ssr: false }
);

import "@bentoproject/accordion/styles.css";
```

For detailed information on dynamic import, refer https://nextjs.org/docs/advanced-features/dynamic-import

I am also trying to figure out root cause of the issue. Will post when found!

**Comment by Beethoven** [comment]
> Hi @Beethoven 👋🏻
> 
> Would you please try using dynamic `import` with `ssr` disabled instead of importing conventionally?
> 
> Syntax:
> 
> ```js
> import dynamic from "next/dynamic";
> 
> const <module_name> = dynamic(() =>
>   import(<path>).then((mod) => mod.<component_to_use>),
>   { ssr: false }
> );
> ```
> 
> So, this import block:
> 
> ```js
> import {
>   BentoAccordion,
>   BentoAccordionSection,
>   BentoAccordionHeader,
>   BentoAccordionContent,
> } from '@bentoproject/accordion/react';
> 
> import '@bentoproject/accordion/styles.css';
> ```
> 
> would be written as:
> 
> ```js
> import dynamic from "next/dynamic";
> 
> // Use dynamic import with SSR disabled for all imports except CSS
> const BentoAccordion = dynamic(() =>
>   import("@bentoproject/accordion/react").then((mod) => mod.BentoAccordion),
>   { ssr: false }
> );
> const BentoAccordionSection = dynamic(() =>
>   import("@bentoproject/accordion/react").then((mod) => mod.BentoAccordionSection),
>   { ssr: false }
> );
> const BentoAccordionHeader = dynamic(() =>
>   import("@bentoproject/accordion/react").then((mod) => mod.BentoAccordionHeader),
>   { ssr: false }
> );
> const BentoAccordionContent = dynamic(() =>
>   import("@bentoproject/accordion/react").then((mod) => mod.BentoAccordionContent),
>   { ssr: false }
> );
> 
> import "@bentoproject/accordion/styles.css";
> ```
> 
> For detailed information on dynamic import, refer https://nextjs.org/docs/advanced-features/dynamic-import
> 
> I am also trying to figure out root cause of the issue. Will post when found!

Yes, this works, but the robots can't see the info on Accordion (the same with all Bento Components) as It's not rendered on server, only on client.
Thus, all Bento components are useless from the SEO perspective.

**Comment by Beethoven** [comment]
And the dos on https://bentojs.dev/documentation/guides/bento-next.js-guide/ is wrong.
Client-side rendering can increase the likelihood of a poor user experience and for search engine bots.
If Bento is for client side only, it's useless.

**Comment by slavakurilyak** [comment]
Reporting the same `self is not defined` error here 👇

![error](https://user-images.githubusercontent.com/6625584/173197052-856303bd-3614-4057-a4ea-bd33a5dab29e.jpg)
